### PR TITLE
docs: add example for accessing vite config programmatically

### PR DESCRIPTION
**Pull request by danielroe**
### 🔗 Linked issue

https://github.com/nuxt/nuxt/issues/14534

### ❓ Type of change

- [x] 📖 Documentation (updates to the documentation or readme)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

This PR adds docs for an example way to access configuration information using Nuxt Kit utilities.

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/3.api/4.advanced/2.kit.md [hunk 1]
```diff
@@ -110,3 +110,39 @@ description: Nuxt Kit provides composable utilities to help interacting with Nux
...
+    buildNuxt(nuxt).catch((err) => {
+      if (!err.toString().includes('_stop_')) {
+        reject(err);
+      }
+    });
+  }).finally(() => nuxt.close());
+}
+
+const viteConfig = await getViteConfig();
+console.log(viteConfig);
```
Comment (DamianGlowala) [inline docs/3.api/4.advanced/2.kit.md:147]:
```suggestion
import { loadNuxt, buildNuxt } from '@nuxt/kit'

// https://github.com/nuxt/nuxt/issues/14534
async function getViteConfig() {
  const nuxt = await loadNuxt({ cwd: process.cwd(), dev: false, ssr: false })
  return new Promise((resolve, reject) => {
    nuxt.hook('vite:extendConfig', (config) => {
      resolve(config)
      throw new Error('_stop_')
    })
    buildNuxt(nuxt).catch((err) => {
      if (!err.toString().includes('_stop_')) {
        reject(err)
      }
    })
  }).finally(() => nuxt.close())
}

const viteConfig = await getViteConfig()
console.log(viteConfig)
```

**Comment by pi0** [comment]
As we are documenting the usage and there are some valid use cases, I believe we should still expose it as a `@nuxt/kit` utility. At least managing and unifying the workarounds behind it.

**Comment by floroz** [comment]
Thank you @danielroe for exposing this documentation.

I am on `nuxt@3.1.2` and I get a TS error for the signature of the option of `loadNuxt`

Screenshot 2023-02-06 at 14 26 00

```sh
Type '{ cwd: string; dev: false; ssr: boolean; }' is not assignable to type 'LoadNuxtOptions'.
  Object literal may only specify known properties, and 'ssr' does not exist in type 'LoadNuxtOptions'.
```

**Comment by danielroe** [comment]
@floroz Thanks for catching that - fixed ✅

**Pull request merged**